src/ui/components/components.rs

```diff
@@ -175,33 +174,35 @@ pub struct Guilds {
     chat_stack: Stack,
     pub guilds_element: gtk4::Box,
     pub discriminator: u8,
+    chat: Rc<RefCell<Chat>>,
 }
 impl Guilds
 {
-    pub(crate) fn new(chat_stack: Stack) -> Self {
+    pub(crate) fn new(chat_stack: Stack, chat: Rc<RefCell<Chat>>) -> Self {
         let guilds_element = gtk4::Box::new(Orientation::Vertical, 0);
-        let dm_box = gtk4::Box::new(Orientation::Vertical, 0);
+        let button_contents = gtk4::Box::new(Orientation::Vertical, 0);
+        button_contents.set_valign(Align::Center);
 
-        let discord_logo = Image::from_file(Path::new("src/ui/assets/discord_logo.png").to_owned());
+        let discord_logo = Image::from_file(Path::new("src/ui/assets/chat_button_logo.png").to_owned());
+        button_contents.append(&discord_logo);
 
-        dm_box.set_valign(Align::Center);
-        dm_box.append(&discord_logo);
+        let chat_button = Button::new();
+        chat_button.set_child(Some(&button_contents));
 
-        let dm_button = Button::new();
-        dm_button.set_child(Some(&dm_box));
-
-        dm_button.connect_clicked(glib::clone!(
-            @weak chat_stack =>
+        chat_button.connect_clicked({
+            let chat_stack = chat_stack.clone();
+            let chat = chat.clone();
```

Comment:
You are cloning the entire chat struct instead off just cloning a reference to the gtk object